app/models/chargeback.rb

```diff
@@ -152,7 +152,7 @@ def calculate_costs(consumption, rates)
     self.class.try(:refresh_dynamic_metric_columns)
 
     rates.each do |rate|
-      rate.rate_details_relevant_to(relevant_fields).each do |r|
+      rate.rate_details_relevant_to(relevant_fields, self.class.attribute_names).each do |r|
```

Comment:
It feels like this should be a method... I believe you're doing a set intersection of `relevant_fields` and  `self.class.attribute_names)` in `rate_details_relevant_to` below, right?

```ruby
def calculate_costs(consumption, rates)
  self.fixed_compute_metric = consumption.chargeback_fields_present if consumption.chargeback_fields_present
  self.class.try(:refresh_dynamic_metric_columns)
	 
  rates.each do |rate|
    rate.rate_details_relevant_to(relevant_fields & self.class.attribute_names).each do |r|
```

Either that or move the set intersection into the method below.

Reply:
Unfortunately , it is not same.

Originally, 
method `rate_details_relevant_to` is going thru [`all rate chargeback details` ](https://github.com/ManageIQ/manageiq/blob/master/app/models/chargeback_rate.rb#L40) and it is basically asking each `rate detail chargeback` - Does your cost calculation affect a selected report fields(report_cols) ?

And it will meet also `rate detail chargeback with derived_vm_numvcpu_cores`.
But if you have only total_cost on report - this total value are  affecting all columns - derived_vm_numvcpu_cores included.

But we need to refuse `derived_vm_numvcpu_cores` for chargeback vm.

Hopefully, you understand better.

And sure we need to do some refactoring here: for example similar condition is also in `ChargebackRateDetail#charge:97` method, I will do it in folluw up PR